fail2ban/tests/files/logs/routeros-auth

```diff
@@ -0,0 +1,9 @@
+
+# failJSON: { "match": true , "host": "192.168.88.6" }
+Feb 15 11:25:46 gw.local system,error,critical login failure for user admin from 192.168.88.6 via web
```

Comment:
See the notice above.

Reply:
Anything special to do here besides fixing the port directive in `jail.conf`?

Reply:
At first look noting...
Although the construction `(system,?|error,?|critical,?)+` looks a bit weird to me.
Are there really constellation where the log doesn't contain `system,error,critical` exactly, as in your tests? The order may change? Some keyword is optional?
Anyway if wanted such a strange dynamic here, I'd rewrite it like that:
```diff
- ... (system,?|error,?|critical,?)+ ...
+ ... (?:,?(?:system|error|critical)){1,3} ...
```
The question is if it really dynamical (free ordered and optional in parts) and not so important, why just not use `\S+` instead. So it would simply ignore up to first space and search `login failure` hereafter.
Otherwise (e. g. the order is fix, but for instance only `system` and `critical` are optional) I'd simply write something like that instead:
```diff
- ... (system,?|error,?|critical,?)+ ...
+ ... (?:system,)?error(?:,critical)? ...
```
This would avoid cyclic branching, thus whole RE becomes faster, especially because of non-greedy match using `.*?`

Reply:
Absolutely no - `.+` is a catch-all, which is:
- really bad before a capturing groups matching foreign input (`.*?` is already bad enough, which is justified because RE is anchored from both sides)
- completely opens a left-anchor - `^.+ login ... $` is almost the same as `login ... $`
- due to unanchored nature makes usage of another catch-all (non-greedy `.*?`) unjustified.

Don't do that, please.

Reply:
Thank you for the feedback. I went through the documentation once again and collected more log samples from all over the internet.
The `system,error,critical` sequence appears identical in all the logs I found, so my original attempt was probably just an over-engineering. I also extended the example log messages for the test purposes.

I pushed a new version to my branch.